### PR TITLE
Revise What's New for inferring compression from non-string paths

### PR DESCRIPTION
**Pull request by dhimmel**
This pull request improves the What's New message corresponding to https://github.com/pandas-dev/pandas/pull/17206, which updated `io.common._infer_compression` to infer compression from non-string paths.

Refs https://github.com/pandas-dev/pandas/issues/17262
Refs https://github.com/pandas-dev/pandas/pull/17206#issuecomment-322586996

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -126,7 +126,7 @@ Other Enhancements
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 - Integration with `Apache Parquet <https://parquet.apache.org/>`__, including a new top-level :func:`read_parquet` and :func:`DataFrame.to_parquet` method, see :ref:`here <io.parquet>`.
 - :func:`DataFrame.add_prefix` and :func:`DataFrame.add_suffix` now accept strings containing the '%' character. (:issue:`17151`)
-- `read_*` methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`).
+- Certain read/write methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`). Only io methods that rely on ``io.common._infer_compression`` are affected, which at this time is ``read_csv``, ``read_table``, ``read_pickle``, and ``write_pickle`` (:issue:`17262`).
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.21.0.txt]: I don't think we should mention the private `io.common._infer_compression` in the whatsnew notes, I would just list the improved functions. (also write_pickle is not a public one -> to_pickle)

File: doc/source/whatsnew/v0.21.0.txt [hunk 2]
```diff
@@ -126,7 +126,7 @@ Other Enhancements
 - :func:`date_range` now accepts 'Y' in addition to 'A' as an alias for end of year (:issue:`9313`)
 - Integration with `Apache Parquet <https://parquet.apache.org/>`__, including a new top-level :func:`read_parquet` and :func:`DataFrame.to_parquet` method, see :ref:`here <io.parquet>`.
 - :func:`DataFrame.add_prefix` and :func:`DataFrame.add_suffix` now accept strings containing the '%' character. (:issue:`17151`)
-- `read_*` methods can now infer compression from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`).
+- Read/write methods that infer compression (``read_csv``, ``read_table``, ``read_pickle``, and ``to_pickle``) can now infer from non-string paths, such as ``pathlib.Path`` objects (:issue:`17206`).
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: can you use `` :func:`read_csv` `` and so on.

**Comment by jreback** [comment]
minor comment. pls rebase. ping on green.

**Comment by jorisvandenbossche** [comment]
Updated and merged. Thanks @dhimmel !

**Pull request merged**